src/epics/chatEpics.js

```diff
@@ -0,0 +1,227 @@
+import {
+    CHAT_INIT_APPLICATION,
+    CHAT_SET_USERNAME,
+    CHAT_INIT_CONNECTION,
+    CHAT_INIT_CHAT,
+    CHAT_SEND_MESSAGE,
+    readyToConnect,
+    pincodeRequired,
+    useParanoidMode,
+    initConnection,
+    connectionInitiated,
+    disconnected,
+    addMessage,
+    setUserStatus,
+    usernameSet
+} from '../actions/chatActions';
+
+import {
+    alertSuccess,
+    alertWarning
+} from '../actions/alertActions';
+
+import {
+    USER_STATUS_DELAY_MS
+} from '../constants/messaging';
+
+import { getEmail, getPassphrase, generateMessageKey } from '../utils/encrypter';
+import ChatHandler from './helpers/ChatHandler';
+import miniLock from '../utils/miniLock';
+import { combineEpics } from 'redux-observable';
+import createDetectVisibilityObservable from './detectPageVisibilityObservable';
+
+const BACKEND_URL = process.env.BACKEND_URL;
```

Comment:
@Spetastian This is old and can be deleted; we're using `window.location.origin` instead so that LeapChat is accessible via multiple URLs at once (e.g., via a combination of an Onion Service (.onion site), localhost, proxies like ngrok, etc).

Reply:
Ill fix it then 😄

Reply:
:+1: